### PR TITLE
PXC-3470: Merge codership/galera 25.3.31 -> percona/galera 3.x

### PR DESCRIPTION
**Pull request by venkatesh-prasad-v**
https://jira.percona.com/browse/PXC-3470
    
***
Merge conflicts resolved.
    
***
Updated GALERA_VER to 3.44.
    
***
Updated GALERA-REVISION to reflect the upstream refresh revision.
    
***
Converted some percona specific unit tests to use `ck_assert*`
libcheck macro instead of the deprecated `fail_if`.
    
***
Reverted our fix for PXC-3424
"PXC-3424: Fix error handling when donor is not able to serve SST"
(https://jira.percona.com/browse/PXC-3424)
(commit 1036b2fd)
in favour of the upstream fix for the bug codership/galera#284
"Fix SST donate callback return value check"
(commit af7cd63d)

Testing Done
---
https://pxc.cd.percona.com/view/PXC%205.6/job/pxc-5.6-param/81/testReport/

Note to Reviewers
---
1. The test case `galera.galera_defaults` is failing because the [recent changes](https://galeracluster.com/2020/10/galera-cluster-for-mysql-5-6-49-5-7-31-and-8-0-21-released). This has been addressed in https://github.com/percona/percona-xtradb-cluster/pull/1309.
```
galera.galera_defaults                   [ fail ]
        Test ended at 2020-11-16 17:12:28

CURRENT_TEST: galera.galera_defaults
--- /tmp/results/PXC/mysql-test/suite/galera/r/galera_defaults.result	2020-11-16 19:21:14.000000000 +0300
+++ /tmp/results/PXC/mysql-test/var/log/galera_defaults.reject	2020-11-16 20:12:27.826260413 +0300
@@ -53,7 +53,7 @@
 WHERE VARIABLE_NAME LIKE 'wsrep_%'
 AND VARIABLE_NAME != 'wsrep_debug_sync_waiters';
 COUNT(*)
-62
+65
 SELECT VARIABLE_NAME FROM INFORMATION_SCHEMA.GLOBAL_STATUS
 WHERE VARIABLE_NAME LIKE 'wsrep_%'
 AND VARIABLE_NAME != 'wsrep_debug_sync_waiters'
@@ -81,12 +81,15 @@
 WSREP_EVS_EVICT_LIST
 WSREP_EVS_REPL_LATENCY
 WSREP_EVS_STATE
+WSREP_FLOW_CONTROL_ACTIVE
 WSREP_FLOW_CONTROL_PAUSED
 WSREP_FLOW_CONTROL_PAUSED_NS
 WSREP_FLOW_CONTROL_RECV
+WSREP_FLOW_CONTROL_REQUESTED
 WSREP_FLOW_CONTROL_SENT
 WSREP_GCACHE_POOL_SIZE
 WSREP_GCOMM_UUID
+WSREP_GMCAST_SEGMENT
 WSREP_INCOMING_ADDRESSES
 WSREP_LAST_COMMITTED
 WSREP_LOCAL_BF_ABORTS

mysqltest: Result content mismatch
```

### CODE REVIEW

File: gcs/src/unit_tests/gcs_group_test.cpp [hunk 1]
```diff
@@ -552,111 +552,55 @@ START_TEST(test_gcs_group_find_donor)
 #define SARGS(s) s, strlen(s)
     //========== sst ==========
     donor = gcs_group_find_donor(&group, sv, joiner, SARGS("home3"),
-<<<<<<< HEAD
                                  &empty_uuid, GCS_SEQNO_ILL, false);
     fail_if(donor != -EHOSTDOWN);
```
Comment (dutow) [inline gcs/src/unit_tests/gcs_group_test.cpp]: wasn't this fail_if changed to ck_assert in upstream? (and a few more below)

File: galera/src/replicator_str.cpp [hunk 2]
```diff
@@ -338,49 +338,6 @@ ReplicatorSMM::donate_sst(void* const         recv_ctx,
     wsrep_cb_status const err(sst_donate_cb_(app_ctx_, recv_ctx,
                                              streq.sst_req(), streq.sst_len(),
                                              &state_id, 0, 0, bypass));
-<<<<<<< HEAD
-
-    /* The fix to codership/galera#284 may break backward comatibility due to
```
Comment (dutow) [inline galera/src/replicator_str.cpp:342]: I don't see codership fixing this differently in the commit history. If I missed it, please mention it in the commit message. And if I didn't, and it really isn't there, this shouldn't be removed.

**Review by dutow — CHANGES_REQUESTED** [review]

File: gcs/src/unit_tests/gcs_group_test.cpp (hunk 1 above)
Comment (venkatesh-prasad-v) [reply gcs/src/unit_tests/gcs_group_test.cpp]: Thanks for catching this. Will fix them in the next revision.

File: galera/src/replicator_str.cpp (hunk 2 above)
Comment (venkatesh-prasad-v) [reply galera/src/replicator_str.cpp:342]: Marked as resolved since we decided to go revert our fix and go with upstream's fix.

**Review by dutow — APPROVED** [review]

**Review by kamil-holubicki — APPROVED** [review]
LGTM,
Please run the full MTR test suite before merging this and the server part.

**Comment by venkatesh-prasad-v** [comment]
Testing Done (with https://github.com/percona/percona-xtradb-cluster/pull/1309)
---
Round 1
Galera suite: https://pxc.cd.percona.com/view/PXC%205.6/job/pxc-5.6-param/82/console
Failing tests: galera_3nodes.galera_join_with_cc_A (Fixed as part of percona/galera#199)

Full suite: https://pxc.cd.percona.com/view/PXC%205.6/job/pxc-5.6-param/83/console

Round 2
Galera and sys_vars: https://pxc.cd.percona.com/view/PXC%205.6/job/pxc-5.6-param/84/console
Failing tests: None

Full suite: https://pxc.cd.percona.com/view/PXC%205.6/job/pxc-5.6-param/85/console [In Progress]
Failing tests

Failing tests:
--- PXC-3279
rpl.rpl_stm_mixed_mts_rec_crash_safe_checksum
rpl.rpl_stm_mixed_mts_rec_crash_safe
rpl.rpl_stm_mixed_mts_rec_crash_safe_small
rpl.rpl_stm_mixed_mts_crash_safe
rpl.rpl_stm_mixed_crash_safe
rpl.rpl_crash_safe_master
rpl.rpl_row_mts_rec_crash_safe
rpl.rpl_row_mts_crash_safe
rpl.rpl_row_crash_safe
rpl.rpl_mts_relay_log_post_crash_recovery
innodb.innodb_bug59641

--- PXC-3288
main.alter_table

--- PXC-3289:
tc-heuristic-recover

--- PXC-3283:
innodb.innodb-autoinc

**Review by kennt-percona — APPROVED** [review]
lgtm

**Pull request merged**